### PR TITLE
Relax the "no recourse" policy for HTTPS-RR-HSTS

### LINKED ISSUE #87: Should the presence of an HTTPS record suppress any certificate warning bypass option?
Elsewhere, HTTPSSVC's redirect was referred to as HSTS and it occurred to me we're missing one of HSTS's properties. It redirects and then it directs the browser to suppress the certificate click through button.

We could do something similar and say that HTTPS connections made off an HTTPSSVC record are assumed to have a competent TLS config and don't get a bypass button.

### PR DESCRIPTION
**Pull request by bemasc**
This change removes the requirement to terminate without
recourse and expands the description of the problem.

Fixes #87

### CODE REVIEW

**Review by MikeBishop — APPROVED** [review]

**Comment by bemasc** [comment]
I've converted this PR to "draft" status due to lack of consensus in #87.

File: draft-ietf-dnsop-svcb-https.md [hunk 1]
```diff
@@ -1072,10 +1072,12 @@ clients MUST NOT place any more trust in this signal than if they
...
-either directly or via the above redirect, the client SHOULD terminate the
-connection if there are any errors with the underlying secure transport, such as
-errors in certificate validation. This aligns with Section 8.4 and Section 12.1
-of {{HSTS}}.
+either directly or via the above redirect, the client MAY terminate the
+connection without user recourse if there are any errors with the underlying
+secure transport.  If the connection has a server certificate for the correct
+name from a trusted issuer, and an error is detected (e.g. the certificate
+is revoked or expired), the client SHOULD terminate without recourse.  This
+is a subset of the requirements in Section 8.4 and Section 12.1 of {{HSTS}}.
```
[the remark this answers is not in the document]
Comment (bemasc) [reply draft-ietf-dnsop-svcb-https.md]: I agree.  I've expanded the explanation here.

**Comment by bemasc** [comment]
I've substantially modified this PR to reflect the apparent consensus in #87.

File: draft-ietf-dnsop-svcb-https.md [hunk 2]
```diff
@@ -1071,11 +1071,15 @@ Because HTTPS RRs are received over an often insecure channel (DNS),
...
+transport, such as an error in certificate validation, some clients
+currently offer a "user recourse" that allows the user to bypass the
+security error and connect anyway.
 When making an "https" scheme request to an origin with an HTTPS RR,
-either directly or via the above redirect, the client SHOULD terminate the
-connection if there are any errors with the underlying secure transport, such as
-errors in certificate validation. This aligns with Section 8.4 and Section 12.1
-of {{HSTS}}.
+either directly or via the above redirect, a client that normally offers
+such a user recourse MAY remove it.  Origins that publish HTTPS RRs
```
Comment (enygren) [inline draft-ietf-dnsop-svcb-https.md:1080]: Is there some wording that doesn't involve the "it" which is a little ambiguous?
Perhaps "a client MAY choose to not provide user recourse".

**Review by enygren — CHANGES_REQUESTED** [review]

File: draft-ietf-dnsop-svcb-https.md (hunk 2 above)
Comment (bemasc) [reply draft-ietf-dnsop-svcb-https.md]: Rephrased

**Review by enygren — APPROVED** [review]

**Pull request merged**